app/Http/Controllers/UserController.php

```diff
@@ -0,0 +1,137 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\User;
+use Illuminate\Support\Facades\Redirect;
+use Spatie\Permission\Models\Role;
+use Illuminate\Support\Facades\Hash;
+use Illuminate\Http\Request;
+
+class UserController extends Controller
+{
+    public function index()
```

Comment:
Não foi adicionado o Middleware no construtor. Neste caso se for acessado diretamente o link ocorrerá erro 500 devido a ausência de um Usuário logado.
```suggestion
    public function __construct()
    {
        $this->middleware('auth');
    }

    public function index()
```